src/main/java/com/kmbl/OrderManagementService/services/OrderService.java

```diff
@@ -2,31 +2,122 @@
 
 import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
 import com.kmbl.OrderManagementService.exceptions.ResourceNotFoundException;
+import com.kmbl.OrderManagementService.models.IMSRequestBody;
+import com.kmbl.OrderManagementService.models.IMSResponseObject;
 import com.kmbl.OrderManagementService.models.Order;
 import com.kmbl.OrderManagementService.models.OrderStatus;
+import com.kmbl.OrderManagementService.models.ResponseObject;
 import com.kmbl.OrderManagementService.models.kafka.CancelOrderMessage;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.kmbl.OrderManagementService.models.OrderItem;
 import com.kmbl.OrderManagementService.repositories.OrderRepository;
+
+import reactor.core.publisher.Mono;
+
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 import java.util.List;
+import java.util.ArrayList;
 
 @Service
 public class OrderService {
 
+    private static final Logger logger = LogManager.getLogger(OrderService.class);
+
+    @Value("${inventoryManagementService.endpoint}")
+    private String inventoryManagementServiceEndpoint;
+    
     private final OrderRepository orderRepository;
+    private final WebClient webClient;
+    private final ObjectMapper objectMapper;
     private final KafkaMessagingService messagingService;
     private final DynamoDBMapper mapper;
-
+    
     @Autowired
-    public OrderService(OrderRepository orderRepository, KafkaMessagingService kafkaMessagingService,DynamoDBMapper mapper) {
+    public OrderService(OrderRepository orderRepository, WebClient.Builder webClientBuilder, ObjectMapper objectMapper, KafkaMessagingService kafkaMessagingService,DynamoDBMapper mapper) {
         this.orderRepository = orderRepository;
+        this.webClient = webClientBuilder.baseUrl(inventoryManagementServiceEndpoint).build();
+        this.objectMapper = objectMapper;
         this.messagingService = kafkaMessagingService;
         this.mapper = mapper;
     }
 
+
     public Order createOrder(Order order) {
-        return orderRepository.save(order);
+        try {
+            List<OrderItem> orderItems = order.getOrderItems();
+            List<IMSRequestBody> imsRequestBodyList = new ArrayList<>();
+    
+            // Convert OrderItems to IMSRequestBody
+            for (OrderItem orderItem : orderItems) {
+                IMSRequestBody imsRequestBody = new IMSRequestBody();
```

Comment:
Create a Constructor in RequestBody with OrderItem as argument, dont use use multiple setter statement